### ISSUE TITLE
[Feature] Better Villages

### ISSUE THREAD

**Issue by Cheaterpaul** [TeamLapen/Vampirism#746]
Maybe there could be a way to strengthen the factions power in a village to make it harder to capture.

Some ideas:
 - each time a village is attacked the strengthen is weakened if the attack wasn't successful
 - the greater the strength the more/better defender are spawning/paroling around
 - strength is increased by placing support blocks for the totem/supply resources
 - village capture processes have a cooldown

can be also addressed and taken into consideration  #497  #174

**Comment by Orange1861** [comment]
Personally, I think getting player made villages to work with the system should be part of this (or even made sooner) since making villages are incredibly fun and important to the idea bigger multiplayer factions (since relying entirely on vanilia villages means this mechanic doesn't exist in places that don't have them).

However, since vanilia player made villages are difficult to work with, perhaps there should be a discussion of moving the mechanic from village capture to capturing villagers?

**Comment by Cheaterpaul** [comment]
Yes, that are two different changes, that can be addressed separately. This issue has not a high priority and i think #694 is likely to be addressed first.

**Comment by Orange1861** [comment]
Makes sense. What are your thoughts on adding a mechanic to capture villagers (flip from vampire to hunter and back) separate from village capturing? That way village capturing would have a less significant effect (since all of the villagers would remain vampire controlled without player intervention).
Also, what are your thoughts on making village capturing connected to the villagers themselves so village flipping is connected to majority of villagers rather than the totem?

**Comment by Cheaterpaul** [comment]
Firstly the idea of converting a vampire villager back to normal has its justification and could be somehow possible, but the problem with that is, that is shouldn't be easy (balance/lore wise).

(talking about what is already possible)
Converting all villagers from normal to vampiric is possible even if a lot of villager would die. So the way from a hunter to a vampire village would be already there.

__But__ from a vampire village to a hunter village it would be a pain, given that converting villager isn't cheap. (not even talking about that if there is a normal villager the vampire villager would turn him again).

**Comment by Cheaterpaul** [comment]
But it could be taken into consideration that the villages strength is also based on the villagers faction.
Means that if the village is a vampire village and there are normal villager it is easier to capture (depending on the percentage) and the other way around

**Comment by Orange1861** [comment]
I agree that it shouldn't be easy to unconvert vampire villagers. I was thinking that it should be similar to uncoverting vanilla MC zombie villagers where uncoverting a lot of them is doable, but expensive and mostly a late game thing.

Also, right now, I noticed when a NPC vampire converts a villager, it's instant and the target villager almost always survives. If it took longer and was more dangerous for the target, it could balance out the more expensive but more reliable counter conversion.

**Comment by Cheaterpaul** [comment]
More thoughtful idea:

Basic idea is that the village has a defense strength. When the strength is depleted the village is overtaken.

usage of the strength:
  - more/better defender are spawning with higher defense strength
  - when defender die the defense strength is decreased

modify the strength:
  - has base strength
  - place support blocks to increase the strength
    - blood infused blocks / blood container
    - garlic core containing items

other strength attributes:
  - strength recovers only to the base strength when decreased
  - when strength drop below the value of the support blocks they are getting destroyed
  - attacker strength is unaffected by village strength
